astropy/convolution/convolve.py

```diff
@@ -296,6 +296,8 @@ def convolve(
         )
     elif array_internal.ndim != kernel_internal.ndim:
         raise Exception("array and kernel have differing number of dimensions.")
+    elif array_internal.size == 0:
+        raise Exception("cannot convolve empty array")
```

Comment:
raising the base `Exception` type isn't recommended (`ValueError` would be more appropriate here I think). It should be possible to refine these exceptions to a more specialised type without breaking user code that might rely on current behaviour since `ValueError` is a subclass of `Exception`, so any code that would already catch `Exception` would also catch `ValueError`.
I'll leave it to reviewers to divide wether such a change is desirable or not and wether it's in scope to this PR or better left as a follow up.

Reply:
:+1: on replacing with a `ValueError` in both cases here.

Reply:
thanks for your feedback ! I'll give it a couple days to see if it raises any eyebrows and if not, I'll push the change !

Reply:
no complaints in 2 days -> pushed !